### ISSUE TITLE
columns-{columns} class no longer applied to Query Loop / Post Template

### ISSUE THREAD

**Issue by justintadlock** [WordPress/gutenberg#51346]
With the change from Flex to the Grid layout with the Query Loop + Post Template blocks in #49050, the `columns-{columns}` class [was removed here](https://github.com/WordPress/gutenberg/pull/49050/files#diff-3deceb62c1cbf003eecba31a4452e0c1977e67a83fd6bb823b99d1692c93b71cL220-R225). This means that theme authors can no longer target queries via custom block styles based on the column number.

Previously, the Post Template block's output would look something like this for a three-column layout:

```html
<ul class="is-flex-container columns-3 wp-block-post-template is-layout-flow wp-block-post-template-is-layout-flow">
```

Today, it looks like this:

```html
<ul class="wp-block-post-template is-layout-grid wp-container-34 wp-block-post-template-is-layout-grid">
```

The new gap support is much needed, so I don't think this needs to be reverted. We just need to add the `columns-{columns}` class back in unless there is a technical reason not to do so (didn't see this mentioned anywhere in the PR).

**Comment by ndiego** [comment]
@tellthemachines @andrewserong any ideas on this one?

**Comment by andrewserong** [comment]
I think it should be possible to add these classnames back in at render time. I'll have a play and try it out! Thanks for raising the issue @justintadlock 👍

**Comment by andrewserong** [comment]
I've opened a potential fix over in #51358.

**Comment by justintadlock** [comment]
Awesome! Great work, @andrewserong.
